### DISCUSSION TITLE
Fastapi with kafka consumer and , db read background process

### DISCUSSION THREAD

**Discussion by aru88** [fastapi/fastapi#5962] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
enpoint i have created 

@etcd_app.get("/api/v1/license", status_code=status.HTTP_200_OK)
def read_license_info(ids: Union[str, None] = None):
    '''
    Returns License info saved in ETCD DB
    Parameters
    - ids (str): Optional parameter that specifies the comma separated
      license key ids for which license info are required.
    Response
    - Successful Response (Status Code: 200)
    - Bad Request (Status Code: 400)
    - {"Message": "An error has occurred."} (Status Code: 500)

    '''

    licencedetails_list = []
    try:
        if not ids:
            #print('prefix:',EtcdConfiguration.prefix)
            itr = read_from_etcd_prefix("/PDA/")
            licencedetails_list = create_license_details_list(itr)
        else:
            license_ids_str = ','.join(f"{e}" for e in ids.split(','))
            for id in license_ids_str.split(","):
                key = create_key(id)
                itr = read_from_etcd_prefix("/PDA/" + key)
                list = create_license_details_list(itr)
                licencedetails_list.extend(list)
        response_dict = {"licenses": licencedetails_list}
        return response_dict
    except ValueError as exc:
        raise HTTPException(
            status_code=status.HTTP_400_BAD_REQUEST,
            detail=str(exc)
        )
    except Exception as exc:
        raise HTTPException(
            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
            detail=str(exc) + ' An error has occurred. '
        )
```

### Description

This is the current code for process in python microservice .
I need to now add one fastapi rest endpoint as well .
is there any option in fast api to run it along with the scheduler and Kafka consumer.

 def start(self):
        try:
            # starting license_consumption_process
            license_consumption_process = Process(
                target=self.listener.run
                )
            # Starting license_reader_process
            license_reader_process = Process(
                target=self.read_license_obj.start_license_info_reader
                )
            license_consumption_process.start()
            license_reader_process.start()
            license_consumption_process.join()
            license_reader_process.join()
        except Exception as e:
            logger.error("exception when starting the process: ", e)

Thanks & regards,
Aruna

### Operating System

Linux

### FastAPI Version

0.75.1

### Python Version

3.8

**Comment by caeser1996** [comment]
It is not possible to add a FastAPI endpoint to this existing code without modifying the code. However, you can add a FastAPI endpoint and still run the scheduler and Kafka consumer alongside it.

First, you need to create a FastAPI application object, add the endpoint to it, and then run the FastAPI app using the run method. You can run this FastAPI app in a separate process using the Process class just like the scheduler and Kafka consumer.

Here's an example:
```
from fastapi import FastAPI
from multiprocessing import Process

app = FastAPI()

@app.get("/api/v1/patient_data")
def get_patient_data():
    # code to get patient data from EHR
    return {"patient_data": patient_data}

def start_fastapi_app():
    app.run()

def start_processes():
    try:
        fastapi_process = Process(target=start_fastapi_app)
        # starting license_consumption_process
        license_consumption_process = Process(
            target=self.listener.run
        )
        # Starting license_reader_process
        license_reader_process = Process(
            target=self.read_license_obj.start_license_info_reader
        )
        fastapi_process.start()
        license_consumption_process.start()
        license_reader_process.start()
        fastapi_process.join()
        license_consumption_process.join()
        license_reader_process.join()
    except Exception as e:
        logger.error("exception when starting the process: ", e)

if __name__ == '__main__':
    start_processes()
```
This way, you can run the FastAPI endpoint alongside the scheduler and Kafka consumer.

**Reply by tiangolo** [reply]
Hey there, it seems you are copying output from ChatGPT or similar, please don't do that, the content generated by those systems is worse than no answer at all, as it not only tends to lie, but also does it in a very convincing way, and that causes more damage than benefit.
